### ISSUE TITLE
Motion planning for grasp trajectory

### ISSUE THREAD

**Issue by LQTS** [kristery/dex-affordance#2]
Hi Yueh-Hua,

Thanks for providing the excellent project.
It seems that I do not find the script of motion planning from a given grasping pose.  Can you provide me with the script of CEM planner?
Thank you very much!

Best,
Quentin Liu

**Comment by maple-zxh** [comment]
Hi,did you solve this problem?

**Comment by LQTS** [comment]
> Hi,did you solve this problem?

No, I have no idea. I tried many other methods, but I can't generate smooth trajectories. Maybe these methods can help you settle your problems down. If you want to try them, you can refer to [https://pregrasps.github.io/](url) and [https://github.com/aravindr93/trajopt](url)

**Comment by maple-zxh** [comment]
Thank you for your reply, I will try them.

**Comment by kristery** [comment]
Hi Quentin Liu,

Thank you for your interest in our work. Here is our script for the cem planning: https://drive.google.com/drive/folders/1Cklymt7A9Ig3wPAw6BhNg8-UD022ZM9B?usp=drive_link

**Comment by maple-zxh** [comment]
Hi Yueh-Hua,

Thank you for the CEM planning script.
If I want to replace it with my own model file, what I should provide is the pose (position, quaternion) of the object model as well as the poses corresponding to the 21 key points of the transformed hand, which are tested under the simulation environment. Is the initial pose of the object set randomly?

Best,
Xuhui

**Comment by binbinbai** [comment]
@kristery Hi, could you provide a joints_dict file used in the CEM planner? The file is loaded with the following code:
`joints_dict = np.load(f'./data/new_joints_shapenet_{category}_scale_1.npy', allow_pickle=True).item()`
Thank you very much!
